packages/next-on-pages/templates/_worker.js/routes-matcher.ts

```diff
@@ -40,6 +40,8 @@ export class RoutesMatcher {
 
 	/** Counter for how many times the function to check a phase has been called */
 	public checkPhaseCounter;
+	/** Tracker for the middleware that have been invoked in a phase */
+	private middlewareInvoked: string[];
```

Comment:
should/cloud this be a flag instead?

as in
```ts
private middlewareInvoked: boolean;
```

and you set it to `true`/`false` as needed?
do we need to check/care that the `middlewarePath` is the same? shouldn't we just prevent multiple middlewares from running regardless?

Reply:
mh... maybe if `middlewarePath` is different we should the different once 🤔

Reply:
I wanted to do something that would allow multiple middlewares on different paths to still get invoked.

Will there ever be a case where there are multiple with different paths? Not sure, maybe not, but that's why I didn't just do a Boolean.

Reply:
It was a fix for the same middleware running multiple times, so I feel like considering the path is somewhat important.

Reply:
yes I was just asking, all of this feels a bit unclear to me in any case 😓